content/articles/geofencing-in-android-with-kotlin/index.md

```diff
@@ -0,0 +1,367 @@
+implementation 'com.google.android.gms:play-services-location:18.0.0'
+```
+
+#### Step 3 - Adding the required permissions
+To begin using geofencing, the user must first check and allow location permissions.
+
+In the Android manifest file, add the following permissions:
+
+```xml
+<uses-permission  android:name="android.permission.ACCESS_FINE_LOCATION" />
+<uses-permission  android:name="android.permission.ACCESS_BACKGROUND_LOCATION" />
+```
+
+#### Check permissions
+Before declaring the function, make sure the app has permission to run in the foreground and background. It's useful to look into the API used by the device.
+
+Perform the following actions:
+
+```kotlin
+private val gadgetQ = android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.Q
+```
+
+To determine whether permission has been granted or not, create the following functions:
+
+```kotlin
+@TargetApi(29)
+private fun approveForegroundAndBackgroundLocation(): Boolean {
+    val foregroundLocationApproved = (
+            PERMISSION_GRANTED == ActivityCompat.checkSelfPermission(
+                this, Manifest.permission.ACCESS_FINE_LOCATION
+            ))
+    val backgroundPermissionApproved =
+        if (gadgetQ) {
+            PERMISSION_GRANTED == ActivityCompat.checkSelfPermission(
+                this, Manifest.permission.ACCESS_BACKGROUND_LOCATION
+            )
+        } else {
+            true
+        }
+    return foregroundLocationApproved && backgroundPermissionApproved
+}
+```
+
+If the device is running Android Q (API 29), ensure that the permissions `ACCESS_BACKGROUND_LOCATION` and `ACCESS_FINE_LOCATION` are enabled. If the device is running an older version, you don't require permission to view the location in the background.
+
+```kotlin
+    private fun authorizedLocation(): Boolean {
+        val formalizeForeground = (
+                PERMISSION_GRANTED == ActivityCompat.checkSelfPermission(
+                    this, Manifest.permission.ACCESS_FINE_LOCATION
+                ))
+        val formalizeBackground =
+            if (gadgetQ) {
+                PERMISSION_GRANTED == ActivityCompat.checkSelfPermission(
+                    this, Manifest.permission.ACCESS_BACKGROUND_LOCATION
+                )
+            } else {
+                true
+            }
+        return formalizeForeground && formalizeBackground
+    }
+}
+```
+
+##### Request background and fine location permissions
+This is where you request permission from the user to access their location if not granted.
+
+```kotlin
+ private fun askLocationPermission() {
+        if (authorizedLocation())
+            return
+        var grantingPermission = arrayOf(Manifest.permission.ACCESS_FINE_LOCATION)
+        val customResult = when {
+            gadgetQ -> {
+                grantingPermission += Manifest.permission.ACCESS_BACKGROUND_LOCATION
+                REQUEST_FOREGROUND_AND_BACKGROUND_PERMISSION_RESULT_CODE
+            }
```

Comment:
I can't seem to find where these codes (Request code and Result code) were initialized 🤭. Kindly assign them some values.